test/mpi/rma/lock_dt_flush.c

```diff
@@ -34,53 +58,74 @@ int main(int argc, char *argv[])
+                sendtype = send_dtp->DTP_obj_array[j].DTP_obj_type;
+                sendbuf = send_dtp->DTP_obj_array[j].DTP_obj_buf;
+
+                if (rank == source) {
                     MPI_Win_lock(MPI_LOCK_SHARED, dest, 0, win);
-                    MPI_Accumulate(sendtype.buf, sendtype.count,
-                                   sendtype.datatype, dest, 0,
-                                   recvtype.count, recvtype.datatype, MPI_REPLACE, win);
+                    MPI_Accumulate(sendbuf, sendcount,
+                                   sendtype, dest, 0, recvcount, recvtype, MPI_REPLACE, win);
                     MPI_Win_flush(dest, win);
                     /*signal to dest that the ops are flushed so that it starts checking the result */
                     MPI_Barrier(comm);
                     /*make sure dest finishes checking the result before issuing unlock */
                     MPI_Barrier(comm);
                     MPI_Win_unlock(dest, win);
 
-                    char *resbuf = (char *) calloc(lb + extent * recvtype.count, sizeof(char));
+                    char *resbuf = (char *) calloc(lb + extent * recvcount, sizeof(char));
 
                     /*wait for the destination to finish checking  and reinitializing the buffer */
                     MPI_Barrier(comm);
 
                     MPI_Win_lock(MPI_LOCK_SHARED, dest, 0, win);
-                    MPI_Get_accumulate(sendtype.buf, sendtype.count,
-                                       sendtype.datatype, resbuf, recvtype.count, recvtype.datatype,
-                                       dest, 0, recvtype.count, recvtype.datatype, MPI_REPLACE,
-                                       win);
+                    MPI_Get_accumulate(sendbuf, sendcount,
+                                       sendtype, resbuf, recvcount, recvtype,
+                                       dest, 0, recvcount, recvtype, MPI_REPLACE, win);
                     MPI_Win_flush(dest, win);
                     /*signal to dest that the ops are flushed so that it starts checking the result */
                     MPI_Barrier(comm);
                     /*make sure dest finishes checking the result before issuing unlock */
                     MPI_Barrier(comm);
                     MPI_Win_unlock(dest, win);
                     free(resbuf);
-                } else if (rank == dest) {
-
+                }
+                else if (rank == dest) {
                     MPI_Barrier(comm);
                     MPI_Win_lock(MPI_LOCK_SHARED, dest, 0, win);
-                    err = MTestCheckRecv(0, &recvtype);
-                    if (err)
+                    err = DTP_obj_buf_check(recv_dtp, i, 0, 1, count);
+                    if (err != DTP_SUCCESS) {
                         errs++;
-                    recvtype.InitBuf(&recvtype);
+                    }
+                    memset(recvbuf, 0, lb + extent * recvcount);
                     MPI_Barrier(comm);
```

Comment:
Please see my previous `memset` comment.

Reply:
this test has been split into `mpi_accumulate` and `mpi_get_accumulate` and `memset` has been removed